src/test/java/com/puppycrawl/tools/checkstyle/gui/ParseTreeTableModelTest.java

```diff
@@ -0,0 +1,404 @@
+                .that(javadoc)
+                .isInstanceOf(DetailNode.class);
+        final int type = ((DetailNode) javadoc).getType();
+        assertWithMessage("Invalid child token type")
+            .that(type)
+            .isEqualTo(JavadocTokenTypes.JAVADOC);
+        final int javadocChildCount = parseTree.getChildCount(javadoc);
+        assertWithMessage("Invalid child count")
+            .that(javadocChildCount)
+            .isEqualTo(5);
+    }
+
+    @Test
+    public void testJavadocChild() {
+        final DetailAST commentContentNode = classDef
+                .getFirstChild().getNextSibling().getFirstChild();
+        final ParseTreeTableModel parseTree = new ParseTreeTableModel(null);
+        parseTree.setParseMode(ParseMode.JAVA_WITH_JAVADOC_AND_COMMENTS);
+        final Object javadoc = parseTree.getChild(commentContentNode, 0);
+        assertWithMessage("Invalid child type")
+                .that(javadoc)
+                .isInstanceOf(DetailNode.class);
+        final int type = ((DetailNode) javadoc).getType();
+        assertWithMessage("Invalid child token type")
+            .that(type)
+            .isEqualTo(JavadocTokenTypes.JAVADOC);
+        final Object javadocChild = parseTree.getChild(javadoc, 2);
+        assertWithMessage("Invalid child type")
+                .that(javadocChild)
+                .isInstanceOf(DetailNode.class);
+        final int childType = ((DetailNode) javadocChild).getType();
+        assertWithMessage("Invalid child token type")
+            .that(childType)
+            .isEqualTo(JavadocTokenTypes.TEXT);
+    }
+
+    @Test
+    public void testGetIndexOfChild() {
+        DetailAST ithChild = classDef.getFirstChild();
+        assertWithMessage("Child must not be null")
+            .that(ithChild)
+            .isNotNull();
+        final ParseTreeTableModel parseTree = new ParseTreeTableModel(null);
+        int index = 0;
+        while (ithChild != null) {
+            final int indexOfChild = parseTree.getIndexOfChild(classDef, ithChild);
+            assertWithMessage("Invalid child index")
+                .that(indexOfChild)
+                .isEqualTo(index);
+            ithChild = ithChild.getNextSibling();
+            index++;
+        }
+
+        final int indexOfChild = parseTree.getIndexOfChild(classDef, new DetailAstImpl());
+        assertWithMessage("Invalid child index")
+            .that(indexOfChild)
+            .isEqualTo(-1);
+    }
+
+    /**
+     * The path to class name in InputJavadocAttributesAndMethods.java.
+     * <pre>
+     * CLASS_DEF
+     *  - MODIFIERS
+     *  - Comment classDef
+     *  - LITERAL_CLASS
+     *  - IDENT -> this is the classDef that holds the class name
+     *  Line number 4 - first three lines are taken by javadoc
+     *  Column 6 - first five columns taken by 'class '
+     *  </pre>
+     */
+    @Test
+    public void testGetValueAt() {
+        classDef = classDef.getLastChild().getPreviousSibling();
```

Comment:
Do not override classadef with other value. Make new variable and name it by what it contains

Reply:
done